### ISSUE TITLE
Gradient Normalisation in attack() return Inf and Nan

### ISSUE THREAD

**Issue by itsraval** [huanranchen/AdversarialAttacks#7]
I was attempting to generate adversarial images using the Common Weakness Attack. I have a few pre-trained models with normalisation.
I was not getting the expected results. I investigated the code and I found out that the gradient normalisation in the attack function can generate inf and nan tensors.

File: attacks/AdversarialInput/CommonWeakness.py
Line: 264, 268

I solved this problem by adding a small epsilon to the denominator.

`
inner_momentum = self.mu * inner_momentum + grad / (torch.norm(grad.reshape(N, -1), p=2, dim=1).view(N, 1, 1, 1) + 1e-12)
`

I was only focused on the Common Weakness Attack, so I did not examine the other functions to see if they had the same problem.
